StyleSheets/StyleSheets/StyleSheet.swift

```diff
@@ -1,29 +1,95 @@
-import UIKit
+import Foundation
 
-public protocol StyleSheetProtocol {
-    associatedtype Element
+/// Represent a bunch of properties to be applied to the given view.
+public struct StyleSheet<View>: Equatable {
+    public typealias Inverse = StyleSheet<View>
+    
+    private var nodes: [PartialKeyPath<View>: ErasedProperty<View>]
 
-    func apply(to element: Element)
-}
+    public init() {
+        nodes = [:]
+    }
 
-extension StyleSheetProtocol {
     @discardableResult
-    public func compose<T>(_ property: ReferenceWritableKeyPath<Self, T>, _ value: T) -> Self {
-        self[keyPath: property] = value
-        return self
+    public func apply(to view: inout View) -> Inverse {
+        defer {
+            nodes.forEach { $0.value.apply(to: &view) }
+        }
+        
+        return StyleSheet().with {
+            $0.nodes = self.nodes.mapValues { $0.current(in: view) }
```

Comment:
I suppose this will only work if a property is changed only once after the view is created or reused (when all changes would be reverted to original values), so the framework should either ensure that, or it should store all inverse stylesheets and apply them in reversed order when preparing view for reuse.

Reply:
@andersio what I mean is that if when configuring view after it was reused (and has value of `value0`) one sets the same property twice, first to `value1` and then to `value2`, on reuse it will be reset to the `value1`, not to `value0`

Reply:
@ilyapuchka Latest fixes have tackled this issue with unit tests attached. It also covers changes at partially overlapping key paths, e.g. `\.layoutMargins.left` and `\.layoutMargins` in any order.